src/corona_hakab_model/generation/geographic_circle.py

```diff
@@ -38,8 +38,15 @@ def generate_agents_ages_and_connections_types(self):
         for agent, age in zip(self.agents, ages):
             agent.age = age
             for connection_type in ConnectionTypes:
+                agent_connection_types = []
                 if np.random.random() < self.data_holder.connection_types_prob_by_age[age][connection_type]:
-                    self.connection_type_to_agents[connection_type].append(agent)
+                    agent_connection_types.append(connection_type)
+            
+            if agent.is_adult() and ConnectionTypes.School in agent_connection_types:
+                agent_connection_types.remove(ConnectionTypes.Work)
```

Comment:
i think you would get an error if ConnectionTypes.Work is not in agent_connection_types.
add a check to the if statement to make sure work is in agent connection types.